### PR TITLE
move check-enforcements back into test/support

### LINKED ISSUE #25454: Stand alone blueprint throws error during unit testing
##### **Overview of the issue**
A stand alone blueprint throws the following error during unit testing, 

```
 FAIL  generators/stand-alone/generator.spec.js > SubGenerator stand-alone of stdalone JHipster blueprint > run
TypeError: You don't seem to have a generator with the name “generator-jhipster-stdalone” installed.
But help is on the way:

You can see available generators via npm search yeoman-generator or via http://yeoman.io/generators/. 
Install them with npm install generator-jhipster-stdalone.

To see all your installed generators run yo --generators. Adding the --help option will also show subgenerators. 

If yo cannot find the generator, run yo doctor to troubleshoot your system.
 ❯ checkGenerator ../../generator-jhipster/node_modules/yeoman-environment/dist/environment-base.js:212:23
 ❯ FullEnvironment.create ../../generator-jhipster/node_modules/yeoman-environment/dist/environment-base.js:254:33
 ❯ JHipsterRunContext.build ../../generator-jhipster/node_modules/yeoman-test/dist/run-context.js:548:26
 ❯ JHipsterRunContext.run ../../generator-jhipster/node_modules/yeoman-test/dist/run-context.js:82:13
 ❯ JHipsterRunContext.run ../../generator-jhipster/dist/testing/helpers.js:181:27

 ❯ generators/stand-alone/generator.spec.js:11:7
```
The above error occurs when using generator-jhipster main. Everything seems ok with release 8.1.0 version.

##### **Motivation for or Use Case**
It should be possible to create and successfully run a stand alone blueprint.

##### **Reproduce the error**
Create a blueprint using the attached .yo-rc.json

##### **Related issues**
None.

##### **Suggest a Fix**
Triage still in progress.

@mshima [this](https://github.com/yeoman/environment/blob/1a307abee6a7b6a26638207fe7c840cc8c06d876/src/environment-base.ts#L360) part is returning undefined, which was changed as part of [this pr](https://github.com/yeoman/environment/pull/515/files#diff-19c45f94cc88c199c379b9ea77874ec03bd7e8d7402ee01652117b32add87e43R360). Prior to this change, the [store.meta](https://github.com/yeoman/environment/blob/1a307abee6a7b6a26638207fe7c840cc8c06d876/src/store.ts#L19) seem to have loaded jhipster-blueprint:** & jhipster:** name spaces which was why the blueprint was discover-able (based on my limited knowledge). Currently I only see that jhipster:** name space only being loaded and hence the findMeta method return undefined. 

Can you please provide some insight and I can try to fix this... Thanks...

##### **JHipster Version(s)**
main branch

##### **JHipster configuration**
<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "additionalSubGenerators": "stand-alone",
    "baseName": "stdalone",
    "cli": true,
    "entities": [],
    "generators": {
      "stand-alone": {
        "command": false,
        "priorities": [
          "initializing",
          "prompting",
          "configuring",
          "composing",
          "writing",
          "writingEntities",
          "postWriting"
        ],
        "sbs": null,
        "written": true
      }
    },
    "jhipsterVersion": "8.1.0",
    "localBlueprint": false,
    "sampleWritten": true,
    "subGenerators": []
  }
}
</pre>
</details>

##### **Environment and Tools**

openjdk version "17.0.8" 2023-07-18
OpenJDK Runtime Environment Temurin-17.0.8+7 (build 17.0.8+7)
OpenJDK 64-Bit Server VM Temurin-17.0.8+7 (build 17.0.8+7, mixed mode, sharing)

git version 2.34.1

node: v20.11.1
npm: 10.2.4

Docker version 25.0.3, build 4debf41

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Browsers and Operating System**

- [X] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by dwarakaprasad**
introduced #25384
fixes #25454

This pr is moving the testing/support/check-enforcements.js into test/support/check-enforcements.js.

---

Please make sure the below checklist is followed for Pull Requests.

- [X] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [X] Tests are added where necessary
- [X] The JDL part is updated if necessary
- [X] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [X] Documentation is added/updated where necessary
- [X] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Review by DanielFran — APPROVED** [review]

File: testing/helpers.ts [hunk 1]
```diff
@@ -171,12 +171,15 @@ class JHipsterRunContext extends RunContext<GeneratorTestType> {
...
+   * @param packagePath overriden package path for the blueprint
    * @returns
    */
-  withParentBlueprintLookup(lookups = ['generators']): this {
+  withParentBlueprintLookup(lookups = ['generators'], packagePath?: string | [string]): this {
     const packageRootParent = join(getPackageRoot(), '..');
     if (basename(packageRootParent) === 'node_modules') {
       this.withLookups([{ packagePaths: [join(packageRootParent, '..')], lookups }] as any);
+    } else if (packagePath) {
+      this.withLookups([{ packagePaths: Array.isArray(packagePath) ? packagePath : [packagePath], lookups }] as any);
```
Comment (mshima) [inline testing/helpers.ts:182]: I don't see a reason for this additional api.
`withLookups({ packagePaths: customPackagePath)`
Is cleaner than:
`withParentBlueprintLookup(undefined, customPackagePath)`

Comment (mshima) [inline testing/helpers.ts:182]:
```suggestion
    } else {
      this.withLookups([{ packagePaths: [process.cwd()], lookups }] as any);
```

Comment (dwarakaprasad) [reply testing/helpers.ts:182]: yeah agreed, I will accept your other suggestion and remove the param from the function and let process.cwd() do its trick.

File: testing/helpers.ts [hunk 2]
```diff
@@ -177,6 +177,8 @@ class JHipsterRunContext extends RunContext<GeneratorTestType> {
     const packageRootParent = join(getPackageRoot(), '..');
     if (basename(packageRootParent) === 'node_modules') {
       this.withLookups([{ packagePaths: [join(packageRootParent, '..')], lookups }] as any);
+    } else {
+      this.withLookups([{ packagePaths: [process.cwd()], lookups }] as any);
```
Comment (mshima) [inline testing/helpers.ts:181]:
```suggestion
      // Try to lookup at current path for linked generator-jhipster.
      this.withLookups([{ packagePaths: [process.cwd()], lookups }] as any);
```

**Review by mshima — APPROVED** [review]

**Pull request merged**